### PR TITLE
WT-8695 Remove file_close_sync config and disallow single file checkpoint

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Comment by quchenhao** [comment]
Not sure if I should change the behaviour in test format or not:

```
    /*
     * Verify can return EBUSY if the handle isn't available. Don't yield and retry, in the case of
     * LSM, the handle may not be available for a long time.
     */
    testutil_check(conn->open_session(conn, NULL, NULL, &session));
    session->app_private = table->track_prefix;
    ret = session->verify(session, table->uri, "strict");
    testutil_assert(ret == 0 || ret == EBUSY);
    testutil_check(session->close(session, NULL));
```

**Comment by keithbostic** [comment]
> Not sure if I should change the behaviour in test format or not. If we keep as it is, I think we are getting ebusy most of the time and it is very unlikely we verify the data.

@quchenhao, I think format has to change -- probably the simplest thing is to add a checkpoint before it does the verify calls, after the load, and after the ops finish.

File: src/txn/txn_ckpt.c
Comment (keithbostic) [inline src/txn/txn_ckpt.c:2107]: I thought this change would remove the `__wt_btree_immediately_durable()` test as well. I think that individually checkpointing logged tables breaks reading from checkpoints (at least), and I don't see any advantage to doing that.

File: src/btree/bt_sync.c
Comment (quchenhao) [reply src/btree/bt_sync.c:156]: Yes, correct. It reduces a lot of cases we return ebusy after even have run a checkpoint.

Comment (quchenhao) [reply src/btree/bt_sync.c:156]: I'll merge code after WT-8771 is merged.

File: src/txn/txn_ckpt.c
Comment (quchenhao) [reply src/txn/txn_ckpt.c:2107]: @hbokhari said we should keep this change. My opinion is can we keep it for now and fix this when we try to get the checkpoint cursor working?

Comment (keithbostic) [reply src/txn/txn_ckpt.c:2107]: @kommiharibabu, can you please explain why you want to keep the `__wt_btree_immediately_durable()` test, specfiically, why the ability to separately checkpoint log-commit files is a good thing?

Comment (quchenhao) [reply src/txn/txn_ckpt.c:2107]: Is it because the data would be the same no matter you checkpointed or not after restart for logged table as log recovery should always run?

Comment (kommiharibabu) [reply src/txn/txn_ckpt.c:2107]: The main reason behind allowing logged tables to let them checkpointed individually is they don't have any difference because of logged after restart as @quchenhao said. Regarding supporting checkpoint cursors on these logged tables we may or may not be required to support the individual checkpoints. I also feel it is better to decide during the support of checkpoint cursors whether we allow individual checkpoints for logged tables.

File: test/csuite/wt2323_join_visibility/main.c
Comment (kommiharibabu) [inline test/csuite/wt2323_join_visibility/main.c:214]: How about writing a `testutil_drop` that can be used everywhere instead of repeating the while logic?

File: test/format/wts.c
Comment (kommiharibabu) [inline test/format/wts.c:543]: Multi-line comment format.

File: test/suite/wttest.py
Comment (kommiharibabu) [inline test/suite/wttest.py:667]: We are not checking for the return type EBUSY in these functions? If the failure occurred due to some other problem it loops forever. Also, it is better to change the function name to specify EBUSY handling. How about the first trying to perform the operation and then calling the checkpoint if it fails?

**Review by kommiharibabu — COMMENTED** [review]
Overall LGTM. A couple of minor test updates.

File: src/txn/txn_ckpt.c
Comment (keithbostic) [reply src/txn/txn_ckpt.c:2107]: @kommiharibabu, @quchenhao: The problem I have is I don't understand why we would WANT to allow logged tables to individually checkpoint. The reason here seems to be more than we CAN allow it, and I agree, but I don't see why we would want to.

It's an edge case we have to document and test, and I can't see any advantage to having an exception to a general rule.

Can someone say why applications might want the ability to (1) individually checkpoint or (2) do exclusive operations on logged tables, while not being able to do so with non-logged tables?

If there's something good here I can see keeping the exception, but right now I can't see the reason we'd want to treat logged and non-logged tables differently.

@quchenhao, speaking of which, we will need some general wording in the documentation about this. I can take a run at that if you'd like, up to you.

Comment (quchenhao) [reply src/txn/txn_ckpt.c:2107]: @keithbostic I'll try to disable the logged tables as well.

**Review by keithbostic — APPROVED** [review]
lgtm, but needs User Guide documentation changes still.

File: test/format/wts.c
Comment (quchenhao) [reply test/format/wts.c:548]: I removed the assert and format starts to fail.

Comment (keithbostic) [reply test/format/wts.c:548]: I'm confused why checkpoint returns `EBUSY`, but I'll take your word for it! :)

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Pull request merged**
